slasher/detection/attestations/spanner.go

```diff
@@ -61,54 +57,48 @@ func (s *SpanDetector) DetectSlashingForValidator(
 	numSpans := uint64(len(s.spans))
 
 	if sp := s.spans[sourceEpoch%numSpans]; sp != nil {
-		minSpan := sp[validatorIdx][0]
+		minSpan := sp[validatorIdx].MinSpan
 		if minSpan > 0 && minSpan < distance {
+			slashableEpoch := sourceEpoch + uint64(minSpan)
+			span := s.spans[slashableEpoch%numSpans][validatorIdx]
 			return &types.DetectionResult{
 				Kind:           types.SurroundVote,
 				SlashableEpoch: sourceEpoch + uint64(minSpan),
+				SigBytes:       span.SigBytes,
 			}, nil
 		}
 
-		maxSpan := sp[validatorIdx][1]
+		maxSpan := sp[validatorIdx].MaxSpan
 		if maxSpan > distance {
+			slashableEpoch := sourceEpoch + uint64(maxSpan)
+			span := s.spans[slashableEpoch%numSpans][validatorIdx]
 			return &types.DetectionResult{
 				Kind:           types.SurroundVote,
-				SlashableEpoch: sourceEpoch + uint64(maxSpan),
+				SlashableEpoch: slashableEpoch,
+				SigBytes:       span.SigBytes,
 			}, nil
 		}
 	}
 
 	if sp := s.spans[targetEpoch%numSpans]; sp != nil {
-		filterNum := sp[validatorIdx][2]
-		if filterNum == 0 {
+		// Check if the validator has attested for this epoch or not.
```

Comment:
We also need to check if it's the same dataroot in order to decide if its double. Doing it in another PR?

Reply:
Yeah, that will be handled by detection. This just makes sure the attestation triggers a hard DB check.